sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Implicits.scala

```diff
@@ -62,4 +85,42 @@ object DataSourceV2Implicits {
       new CaseInsensitiveStringMap(options.asJava)
     }
   }
+
+  implicit class TablePartitionSpecHelper(partSpec: TablePartitionSpec) {
+    def asPartitionIdentifier(partSchema: StructType): InternalRow = {
+      val conflictKeys = partSpec.keys.toSeq.diff(partSchema.map(_.name))
+      if (conflictKeys.nonEmpty) {
+        throw new AnalysisException(
+          s"Partition key ${conflictKeys.mkString(",")} not exists")
+      }
+
+      val partValues = partSchema.map { part =>
+        val partValue = partSpec.get(part.name).orNull
+        if (partValue == null) {
+          null
+        } else {
+          part.dataType match {
```

Comment:
How about other types like `DateType`. If you are not going to support them in this PR, please, open an JIRA and add `TODO` here.

Reply:
a `TODO` sounds good to me.